polylogue/cli/maintain.py

```
@@ -20,6 +24,42 @@ def run_maintain_cli(args: argparse.Namespace, env: CommandEnv) -> None:
+        if not src.exists() or not src.is_dir():
+            raise SystemExit(f"Snapshot directory not found: {src}")
+        if dest.exists():
+            if not force:
+                raise SystemExit(f"Destination already exists: {dest} (use --force to overwrite)")
+            shutil.rmtree(dest)
+        dest.parent.mkdir(parents=True, exist_ok=True)
+
+        # Estimate disk need (roughly size of src)
+        try:
+            total_bytes = sum(p.stat().st_size for p in src.rglob("*"))
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)